src/appleseed/renderer/global/globaltypes.h

```diff
@@ -58,6 +58,7 @@ typedef foundation::AABB<GScalar, 3> GAABB3;
 typedef foundation::AABB<GScalar, 1> GAABB1;
 typedef foundation::Ray<GScalar, 3> GRay3;
 typedef foundation::RayInfo<GScalar, 3> GRayInfo3;
+typedef foundation::Color<GScalar, 3> GColor3;
```

Comment:
I don't think this typedef makes sense.

First, a clarification as it may not be obvious: the `G` in all these type names stands for _geometry_. They define with which precision we store geometry in memory.

I believe that we will never ever want to store colors with more precision than what float offers. Thus I'd advocate for always using `foundation::Color3f` for colors, and removing this typedef.

Reply:
Fair enough. Only reason I chose to use this was because I thought GColor3 would only be associated with per-vertex-color, but it is a bit of overkill I agree.